File: frontend/src/types/market.ts
export interface MarketAnnouncementItem {
  article_id: string
  title: string
  category: string | null
  published_at: string | null
  url: string | null
  source: string
}

export interface MarketSnapshotResponse {
  symbol: string
  name: string
  latest_price: number
  change_amount: number | null
  change_percent: number | null
  open_price: number | null
  high_price: number | null
  low_price: number | null
  volume: number | null
  amount: number | null
  total_market_cap: number | null
  main_net_inflow: number | null
  source: string
  as_of: string
  fetched_at: string
  freshness: 'fresh' | 'cached' | 'stale'
  warning: string | null
  announcements: MarketAnnouncementItem[]
}

File: backend/app/schemas/market.py
from datetime import datetime

from pydantic import BaseModel, Field


class MarketAnnouncementItem(BaseModel):
    article_id: str
    title: str
    category: str | None = None
    published_at: datetime | None = None
    url: str | None = None
    source: str


class MarketSnapshotResponse(BaseModel):
    symbol: str = Field(pattern=r"^\d{6}$")
    name: str
    latest_price: float
    change_amount: float | None = None
    change_percent: float | None = None
    open_price: float | None = None
    high_price: float | None = None
    low_price: float | None = None
    volume: float | None = None
    amount: float | None = None
    total_market_cap: float | None = None
    main_net_inflow: float | None = None
    source: str
    as_of: datetime
    fetched_at: datetime
    freshness: str
    warning: str | None = None
    announcements: list[MarketAnnouncementItem]
